### DISCUSSION TITLE
Can I download debug symbols for native DLLs for a particular release?

### DISCUSSION THREAD

**Discussion by lostmsu** [LWJGL/lwjgl3#954] [Q&A] [answered]
I am debugging a 3rd party app that ships with LWJGL 3.3.1 and uses OpenAL.dll. The app hangs somewhere in OpenAL code which I can see in Visual Studio, but I can't see the function names or see the source code because there are no debug symbols included.

Where can I get debug symbols for the official builds of OpenAL included with LWJGL?

**Comment by Spasi** [comment, marked as answer]
Hey @lostmsu,

There are no debug symbols available for the official builds. The recommendation is to replace the official binary with a debug build (see `Configuration.OPENAL_LIBRARY_NAME` or `-Dorg.lwjgl.openal.libname`). LWJGL 3.3.1 was released with OpenAL Soft at [this commit](https://github.com/kcat/openal-soft/commit/ae4eacf147e2c2340cc4e02a790df04c793ed0a9), you can clone it at that point and use [this workflow](https://github.com/LWJGL-CI/openal-soft/blob/master/.github/workflows/lwjgl.yml) as reference to build it (make sure to change the CMake build type option from release to debug).

You may also want to try `ALSOFT_LOGLEVEL` first, see if it produces anything that might help with identifying the issue. More details in [env-vars.txt](https://github.com/kcat/openal-soft/blob/master/docs/env-vars.txt).
